packages/core/adder/preconditions.ts

```diff
@@ -86,32 +86,34 @@ export async function validatePreconditions<Args extends OptionDefinition>(
                 message = precondition.name + ` (Unexpected failure: ${errorString})`;
             }
 
-            if (multipleAdders) {
-                message = `${name}: ${message}`;
-            }
+            if (!preconditionPassed) {
+                if (multipleAdders) {
+                    message = `${name}: ${message}`;
+                }
 
-            message = preconditionPassed ? pc.green(message) : pc.yellow(message);
-            preconditionLog.push(message);
+                message = pc.yellow(message);
+                preconditionLog.push(message);
+            }
 
             if (!preconditionPassed) allPreconditionsPassed = false;
         }
     }
-    if (preconditionLog.length > 0) {
-        let allMessages = "";
-        for (const [i, message] of preconditionLog.entries()) {
-            allMessages += `- ${message}${i == preconditionLog.length - 1 ? "" : "\n"}`;
-        }
 
-        if (!allPreconditionsPassed && isTesting) {
-            throw new Error(`Preconditions failed: ${preconditionLog.join(" / ")}`);
-        }
-        if (isTesting) return;
+    if (allPreconditionsPassed) {
+        return;
+    } else if (!allPreconditionsPassed && isTesting) {
+        throw new Error(`Preconditions failed: ${preconditionLog.join(" / ")}`);
+    }
+
+    let allMessages = "";
+    for (const [i, message] of preconditionLog.entries()) {
+        allMessages += `- ${message}${i == preconditionLog.length - 1 ? "" : "\n"}`;
+    }
```

Comment:
I find this to be a bit easier to parse, though it may just be a preference:
```suggestion
    const allMessages = preconditionLog.map((msg) => `- ${msg}`).join("\n");
```

Reply:
Don't know what my brain was thinking there, you are 150% right!